addons/website_slides/models/slide_channel.py

```diff
@@ -131,6 +163,12 @@ def unlink(self):
             self.env['slide.slide.partner'].search(removed_slide_partner_domain).unlink()
         return super(ChannelUsersRelation, self).unlink()
 
+    def _get_invitation_hash(self):
+        """ Returns the invitation hash of the attendee, used to access courses as invited / joined. """
+        self.ensure_one()
+        token = (self.partner_id.id, self.channel_id.id)
+        return tools.hmac(self.env(su=True), 'website_slides-channel-invite', token)
+
     def _set_as_completed(self, completed=True):
```

Comment:
Isn't it more a "_post_completion_update_hook" now that the completed if computed by "_recompute_completion" only ?